v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go

```diff
@@ -53,29 +56,71 @@ type ResourceVersionsReport struct {
 	objectModelConfiguration *config.ObjectModelConfiguration
 	rootUrl                  string
 	samplesPath              string
-	groups                   set.Set[string]                       // A set of all our groups
-	kinds                    map[string]astmodel.TypeDefinitionSet // For each group, the set of all available resources
-	frontMatter              string                                // Front matter to be inserted at the top of the report
-	// A separate list of resources for each package
-	lists map[astmodel.PackageReference][]astmodel.TypeDefinition
+	frontMatter              string                                                  // Front matter to be inserted at the top of the report
+	availableFragments       map[string]string                                       // A collection of the fragments to use in the report
+	groups                   set.Set[string]                                         // A set of all our groups
+	kinds                    map[string]astmodel.TypeDefinitionSet                   // For each group, the set of all available resources
+	lists                    map[astmodel.PackageReference][]astmodel.TypeDefinition // A separate list of resources for each package
+	typoAdvisor              *config.TypoAdvisor                                     // Advisor used to troubleshoot unused fragments
 }
 
 func NewResourceVersionsReport(
 	definitions astmodel.TypeDefinitionSet,
 	cfg *config.Configuration,
-) *ResourceVersionsReport {
+) (*ResourceVersionsReport, error) {
 	result := &ResourceVersionsReport{
 		reportConfiguration:      cfg.SupportedResourcesReport,
 		objectModelConfiguration: cfg.ObjectModelConfiguration,
 		rootUrl:                  cfg.RootURL,
 		samplesPath:              cfg.SamplesPath,
+		availableFragments:       make(map[string]string),
 		groups:                   set.Make[string](),
 		kinds:                    make(map[string]astmodel.TypeDefinitionSet),
 		lists:                    make(map[astmodel.PackageReference][]astmodel.TypeDefinition),
+		typoAdvisor:              config.NewTypoAdvisor(),
+	}
+
+	err := result.loadFragments()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to load report fragments")
 	}
 
 	result.summarize(definitions)
-	return result
+	return result, nil
+}
+
+// loadFragments scans the files in the fragments directory and loads them into the availableFragments map
+func (report *ResourceVersionsReport) loadFragments() error {
+	fragmentsPath := report.reportConfiguration.FullFragmentPath()
+
+	files, err := ioutil.ReadDir(fragmentsPath)
```

Comment:
consider using `filepath.Walk` here? It'll make the code cleaner and we won't need a `ReadDir` and a `for` loop seperately.

Reply:
`err := filepath.Walk("fragmentsPath", func(path string, info fs.FileInfo, err error) error {
		if !info.IsDir() {
			content, err := ioutil.ReadFile(path)
			if err != nil {
				return errors.Wrapf(err, "Unable to read fragment file %q", info.Name())
			}

			// Strip the extension from the filename
			name := strings.TrimSuffix(info.Name(), filepath.Ext(info.Name()))
			report.availableFragments[name] = string(content)
		}
		return nil
	})
	if err != nil {
		return errors.Wrapf(err, "Unable to read fragments from directory %q", fragmentsPath)
	}`

Reply:
err := filepath.Walk("fragmentsPath",
		func(path string, info fs.FileInfo, err error) error {
	
			if !info.IsDir() {
				content, err := ioutil.ReadFile(path)
				if err != nil {
					return errors.Wrapf(err, "Unable to read fragment file %q", info.Name())
				}
				// Strip the extension from the filename
				name := strings.TrimSuffix(info.Name(), filepath.Ext(info.Name()))
				report.availableFragments[name] = string(content)
			}

			return nil
		})

	if err != nil {
		return errors.Wrapf(err, "Unable to read fragments from directory %q", fragmentsPath)
	}

Reply:
Seems to work - thanks.